presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -711,6 +711,14 @@ protected Object visitComparisonExpression(ComparisonExpression node, Object con
             }
 
             Object right = process(node.getRight(), context);
+            if (right == null && operator != ComparisonExpression.Operator.IS_DISTINCT_FROM) {
+                return null;
+            }
+
+            if ((left != null && hasUnresolvedValue(left)) || (right != null && hasUnresolvedValue(right))) {
+                return new ComparisonExpression(operator, toExpression(left, type(node.getLeft())), toExpression(right, type(node.getRight())));
+            }
+
             if (operator == ComparisonExpression.Operator.IS_DISTINCT_FROM) {
```

Comment:
Is this block still needed? At this stage we know left & right are both values (not expressions), so we can just rely on `invokeOperator` below